### DISCUSSION TITLE
netcdfAll-5.5.3.jar   Unable to release handle after reading file

### DISCUSSION THREAD

**Discussion by breezePeak** [Unidata/netcdf-java#1257] [Questions and Answers]
try (NetcdfFile ncfile = NetcdfFiles.open(path)) {
            // Do cool stuff here
        } catch (IOException ioe) {
            ioe.printStackTrace();
        }
        System.out.println("111111");   
        Break point on the last line，After starting debugging，Attempting to delete the read in file, Win11 shows that it cannot be deleted,Why is this

**Comment by mnlerman** [comment]
Hello, I wasn't able to reproduce this problem. Admittedly, I'm running a Mac so this wasn't an apples to apples test. I'll keep investigating to see if it is Windows specific problem. In the meantime, could you tell me a few things: 

Are you able to delete the file when your code isn't running? 
Are you deleting the file by hand or from within your code? 
If you add `ncfile.close()` to try end of the try block (which should be redundant since you're using a try-with resource) are you able to delete the file? 
What is the wording of the error message that you get when you try to delete the file?

**Reply by breezePeak** [reply]
I found an issue while parsing grib1. After checking the source code, it should be. When determining the file type, there was an implementation class that did not release the file handle. I have already avoided this issue and resolved it

**Reply by tdrwenski** [reply]
Glad to hear you got your code working. Could you point us to the implementation class that is not releasing the file handle so we can have a look at fixing it?

**Reply by skypanda100** [reply]
you could try to parse grib1 file by using open method which below, and then set the debug point below ```netcdfFile.close();```, then you will find the process didn't release the grib file handle
```java
NetcdfFiles.open(gribFilePath);
```

**Reply by skypanda100** [reply]
## AreaReader.class
```java
    public static boolean isValidFile(RandomAccessFile raf) {
        String fileName = raf.getLocation();
        AreaFile af = null;

        boolean var4;
        try {
            af = new AreaFile(fileName);
            boolean var3 = true;
            return var3;
        } catch (AreaFileException var8) {
            var4 = false;
        } finally {
            if (af != null) {
                af.close();
            }

        }

        return var4;
    }
```
after ```new AreaFile(fileName);```, this function just return true directly, there is no place to call close method!
I found a way to avoid this problem:
```java
NetcdfFiles.open(gribFilePath, "ucar.nc2.grib.collection.Grib1Iosp", 100, null, null);
```

**Reply by tdrwenski** [reply]
Thank you for the extra info! Can you also share the file you are testing with (or a link to where I can download it if it is too large to attach)?

**Comment by skypanda100** [comment]
> Thank you for the extra info! Can you also share the file you are testing with (or a link to where I can download it if it is too large to attach)?

[test.txt](https://github.com/Unidata/netcdf-java/files/13385987/test.txt)
The test.txt is a grib1 file, I changed the suffix for uploading it.

**Reply by tdrwenski** [reply]
Thanks!

**Reply by tdrwenski** [reply]
Thanks again for pointing us to where the issue is! It should be fixed in the latest snapshot version which you can download here [netcdfAll-5.5.4-SNAPSHOT.jar](https://downloads.unidata.ucar.edu/netcdf-java/5.5.4/netcdfAll-5.5.4-SNAPSHOT.jar)

**Reply by skypanda100** [reply]
🤝
